android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -117,6 +120,47 @@ public void onDismiss() {
         setUpViews();
     }
 
+    public void showPopupMenu(){
+       PopupMenu menu = new PopupMenu(mOptionsImage.getContext(),   (View) mOptionsImage);
+       menu.getMenuInflater().inflate(R.menu.menu_dapps_panel, menu.getMenu());
+       menu.setOnMenuItemClickListener(item -> {
+            if (item.getItemId() == R.id.action_lock_wallet) {
+                mBraveWalletPanelServices.getKeyringService().lock();
+                dismiss();
+            }
+            if (item.getItemId() == R.id.action_connected_sites) {
+                //TODO needs to be implemented in another ticket
+                dismiss();
+            }
+            if (item.getItemId() == R.id.action_settings) {
+                BraveActivity activity = BraveActivity.getBraveActivity();
+                if (activity != null) {
+                    activity.openBraveSettings();
+                    dismiss();
+                }
+            }
+            if (item.getItemId() == R.id.action_view_on_block_explorer) {
+                BraveActivity activity = BraveActivity.getBraveActivity();
+                if (activity != null) {
+                    mBraveWalletPanelServices.getKeyringService().getKeyringInfo(
+                    BraveWalletConstants.DEFAULT_KEYRING_ID, keyringInfo -> {
+                        if (keyringInfo != null) {
+                            mAccountInfos = keyringInfo.accountInfos;
+                            if (mAccountInfos.length > 0)
+                                activity.viewOnBlockExplorer(mAccountInfos[0].address);
+                            dismiss();
+                        }
+                    });
+                }
+            }
+            return true;
+        });
+        if (menu.getMenu() instanceof MenuBuilder) {
+            ((MenuBuilder) menu.getMenu()).setOptionalIconsVisible(true);
+        }
```

Comment:
[setOptionalIconsVisible](https://stackoverflow.com/a/66258835) has `RestrictedApi` access so not sure if it is safe/durable to access it, probably a safer option is to use [`MenuPopupHelper`](https://stackoverflow.com/a/39026923)

Reply:
MenuPopupHelper also has RestrictedApi
MenuPopupHelper can only be called from within the same library group prefix (referenced groupId=androidx.appcompat with prefix androidx from groupId=Experiments2)
setOptionalIconVisible() also restricted api...
should we use reflection to fix it like [this](https://stackoverflow.com/a/46265894)

Reply:
Then we can keep the current changes.